src/stavesection.ts

```diff
@@ -51,32 +52,32 @@ export class StaveSection extends StaveModifier {
   }
 
   draw(stave: Stave, shift_x: number): this {
+    const borderWidth = 2;
+    const padding = 2;
     const ctx = stave.checkContext();
     this.setRendered();
 
     ctx.save();
-    ctx.setLineWidth(2);
+    ctx.setLineWidth(borderWidth);
     ctx.setFont(this.textFont);
+    const textFormatter = TextFormatter.create(this.textFont);
 
-    const paddingX = 2;
-    const paddingY = 2;
-    const rectWidth = 2;
-    const textMeasurements = ctx.measureText(this.section);
-    const textWidth = textMeasurements.width;
-    const textHeight = textMeasurements.height;
-    const width = textWidth + 2 * paddingX; // add left & right padding
-    const height = textHeight + 2 * paddingY; // add top & bottom padding
+    const textWidth = textFormatter.getWidthForTextInPx(this.section);
+    const textY = textFormatter.getYForStringInPx(this.section);
+    const textHeight = textY.height;
+    const headroom = -1 * textY.yMin;
+    const width = textWidth + 2 * padding; // add left & right padding
+    const height = textHeight + 2 * padding; // add top & bottom padding
 
     //  Seems to be a good default y
     const y = stave.getYForTopText(1.5) + this.shift_y;
     const x = this.x + shift_x;
     if (this.drawRect) {
       ctx.beginPath();
-      ctx.setLineWidth(rectWidth);
-      ctx.rect(x, y + textMeasurements.y - paddingY, width, height);
+      ctx.rect(x, y - height + headroom, width, height);
       ctx.stroke();
     }
-    ctx.fillText(this.section, x + paddingX, y);
+    ctx.fillText(this.section, x + padding, y - padding);
```

Comment:
y padding has been removed

Reply:
Since the width/height is coming from the actual text and not the max glyph, you need to add padding when placing the text for both x and y.  The width and height from the rectangle already account for the padding.  'headroom' is added for text that drops below the baseline, since 'y' in the text element always refers to the baseline.  So the character 'C' for instance usually drops a bit below the baseline.

Reply:
But you are using 
`const height = textHeight + 2 * padding;`
will not all the padding be left above?

Reply:
The padding * 2 is on the rectangle, because the padding goes on top and bottom, left and right.  But when placing the text, you only consider padding * 1 since text element always places in the lower left.

Visually, there is more space on the top of the boxes because that's how the fonts were designed.  Whether you are measuring or getting the height from the font metrics, the max y tends to return higher than the highest point on the glyph.